### PR TITLE
Slack block kit builder action

### PR DESCRIPTION
**Pull request by dannyroosevelt**
### <samp>🤖 Generated by Copilot at 48e4aa0</samp>

This pull request adds two new action components to the Slack app: `build-block-kit-message` and `build-send-block-kit-message`. These components allow users to create and send custom Slack messages using the Block Kit API. The pull request also adds a common module `build-blocks` that contains the shared logic for building blocks. Additionally, the pull request updates the `pnpm-lock.yaml` file to include a new dependency.

### <samp>🤖 Generated by Copilot at 48e4aa0</samp>

> _`Slack` app gets new blocks_
> _Build and send custom messages_
> _Autumn of updates_

### <samp>🤖 Generated by Copilot at 48e4aa0</samp>

*  Add two new action components to the Slack app for building and sending custom messages using the Block Kit API ([link](https://github.com/PipedreamHQ/pipedream/pull/8278/files?diff=unified&w=0#diff-762d1a5a99bb9378fe66550fbbea71a8c08b2d0dc469c57fd30471bf6c981ed5R1-R154), [link](https://github.com/PipedreamHQ/pipedream/pull/8278/files?diff=unified&w=0#diff-17e899253bd6cf15daab332616f95e156b5feb40e140f37bc59d0adc244e5393R1-R42))
* Extract the common props and methods for building blocks into a separate module in the common folder ([link](https://github.com/PipedreamHQ/pipedream/pull/8278/files?diff=unified&w=0#diff-6a253aebdc394b6b71c1ae11b180f267d9a533ef28c2dd6114ec7406f73a72b6R1-R151))
* Update the `pnpm-lock.yaml` file to include the new dependency on the `components/clickfunnels` module ([link](https://github.com/PipedreamHQ/pipedream/pull/8278/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR849-R850))

### CODE REVIEW

File: components/slack/actions/common/build-blocks.mjs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    const sectionDescription = "Add a **section** block to your message using plain text or mrkdwn. See [Slack's docs](https://api.slack.com/reference/block-kit/blocks?ref=bk#section) for more info.";
+    const contextDescription = "Add a **context** block to your message using plain text or mrkdwn, or specify multiple elements to pass as an array in the context block. See [Slack's docs](https://api.slack.com/reference/block-kit/blocks?ref=bk#context) for more info.";
+    const linkButtonDescription = "Add a **link button** to your message. Enter the button text as the key and the link URL as the value. Configure multiple buttons in the array to render them inline, or add additional Button Link blocks to render them vertically. See [Slack's docs](https://api.slack.com/reference/block-kit/blocks?ref=bk#actions) for more info.";
+    const propsSection = this.createBlockProp("string", "Section Block", sectionDescription);
+    const propsContext = this.createBlockProp("string[]", "Context Block", contextDescription);
+    const propsLinkButton = this.createBlockProp("object", "Link Button", linkButtonDescription);
+
+    if (this.passArrayOrConfigure == "array") {
```
Comment (dannyroosevelt) [inline components/slack/actions/common/build-blocks.mjs:92]: @andrewjschuang could especially use your eyes here, I can't quite figure out how to get this to work

**Review by lcaresia — DISMISSED** [review]
LGTM!

File: components/slack/actions/common/build-blocks.mjs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+          label: "Configure blocks individually",
+          value: "configure",
+        },
+      ],
+      reloadProps: true,
+    },
+  },
+  methods: {
+    // This adds a visual separator in the props form between each block
+    separator() {
```
Comment (andrewjschuang) [inline components/slack/actions/common/build-blocks.mjs:27]: Interesting use!

Comment (dannyroosevelt) [reply components/slack/actions/common/build-blocks.mjs:27]: Inspired by Mr TFrank himself

**Pull request merged**